driver-core/src/main/com/mongodb/connection/LevelCountingBsonWriter.java

```diff
@@ -29,256 +30,256 @@
 
 
 abstract class LevelCountingBsonWriter implements BsonWriter {
-    private final BsonWriter bsonWriter;
+    private final BsonBinaryWriter bsonBinaryWriter;
```

Comment:
To avoid introducing the overloaded pipe method to the BsonWriter interface, changed this to depend on BsonBinaryWriter.